test/apple/code-tools.test.ts

```diff
@@ -1,20 +1,1038 @@
+  const tempDir = fs.mkdtempSync(path.join(os.tmpdir(), 'code-tools-test'));
+  return tempDir;
+};
+
+const prepareAppDelegateFile = (
+  dir: string,
+  content: string,
+  ext: 'm' | 'mm' | 'swift',
+): string => {
+  const filePath = path.join(dir, `AppDelegate.${ext}`);
+  fs.writeFileSync(filePath, content, 'utf8');
+  return filePath;
+};
+
+const dsn = 'https://example.com/sentry-dsn';
+
+// Mock Setup
+
+jest.mock('../../src/utils/bash');
+jest.spyOn(Sentry, 'setTag').mockImplementation();
+jest.spyOn(Sentry, 'captureException').mockImplementation();
+
+// Test Suite
 
 describe('code-tools', () => {
-  describe('addCodeSnippetToProject', () => {
+  beforeEach(() => {
+    jest.spyOn(clack.log, 'info').mockImplementation();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('#isAppDelegateFile', () => {
+    const prepareTestFile = (
+      content: string,
+      ext: 'm' | 'mm' | 'swift',
+    ): string => {
+      const tempDir = prepareTempDir();
+      return prepareAppDelegateFile(tempDir, content, ext);
+    };
+
+    describe('swift files', () => {
+      describe('swift app launch regex', () => {
+        describe('valid cases', () => {
+          const variations: {
+            name: string;
+            code: string;
+          }[] = [
+            {
+              name: 'with underscores',
+              code: 'func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {',
+            },
+            {
+              name: 'with different dictionary type',
+              code: 'func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {',
+            },
+            {
+              name: 'with extra whitespace',
+              code: '  func  application  (  _  application:  UIApplication  ,   didFinishLaunchingWithOptions   launchOptions:   [  NSObject  :  AnyObject  ]?  )   ->   Bool   {  ',
+            },
+            {
+              name: 'macOS notification variant',
+              code: 'func applicationDidFinishLaunching(_ aNotification: Notification) {',
+            },
+            {
+              name: 'macOS with extra whitespace',
+              code: 'func   applicationDidFinishLaunching  (  _   aNotification:  Notification  )  {',
+            },
+          ];
+
+          for (const variation of variations) {
+            describe(`${variation.name}`, () => {
+              it(`should return true`, () => {
+                // -- Arrange --
+                const filePath = prepareTestFile(variation.code, 'swift');
+
+                // -- Act --
+                const result = exportForTesting.isAppDelegateFile(filePath);
+
+                // -- Assert --
+                expect(result).toBeTruthy();
+              });
+            });
+          }
+
+          describe('invalid cases', () => {
```

Comment:
`m`: What about adding some cases for garbage content in the files, such as `adsfasdf` or what if there is a missing `}` in the app delegate, so the code doesn't compile.

Reply:
Good feedback, it seems I made a copy-paste error here (so many tests 😅). I'll add a garbage example example but we actually don't care about compiling here, because we are not interpreting code, only character matching

Reply:
I also don't care too much about compiling here, but I care about the wizard not crashing when somebody has garbage in their AppDelegate 😄

Reply:
I agree, but there is no difference in the test equivalence classes, so it would just result in multiple tests testing for the same thing.

I added something for completeness sake, but just wanted to point this out.

Reply:
I disagree, the `asdasdf garbage` test is an extra equivalence class when you don't know the code. When you know the code it's not, though. I think it adds value. Thanks.